advanced.md

```diff
@@ -167,4 +167,112 @@ public void toggleAllDigitalPins( bool setPinsHigh )
+
+```c#
+public class AnalogWriteAndReport
+{
+	IStream connection;
+	UwpFirmata firmata;
+	RemoteDevice arduino;
+
+	//these volatile variables allow me to terminate my threads safely when necessary.
+	public volatile bool writerThreadShouldExit;
+	public volatile bool readerThreadShouldExit;
+
+	public AnalogWriteAndReport()
+	{
+		//first, we construct our objects. I'm using NetworkSerial to connect to my Arduino over a LAN
+		connection = new NetworkSerial( new Windows.Networking.HostName( "192.168.0.113" ), 5000 );
+		firmata = new UwpFirmata();
+		arduino = new RemoteDevice( firmata );
+
+		//attach the connection to the UwpFirmata object with begin()
+		firmata.begin( connection );
+
+		//attach our ConnectionEstablished callback so we can be informed that the device is ready!
+		connection.ConnectionEstablished += OnNetworkConnectionEstablished;
+
+		//finally, tell the connection to begin. These arguments (baud and serialconfig) don't matter for the NetworkSerial class,
+		//but when they do matter, I like to use 115200 baud rate, and 8N1 is necessary when using USB connections.
+		connection.begin( 115200, SerialConfig.SERIAL_8N1 );
+	}
+
+	//this function is called when the connection is established, and the device is ready.
+	private void OnNetworkConnectionEstablished()
+	{
+		//I'm going to set pins 9 and 10 to PWM mode, so that I can write analog values to them
+		arduino.pinMode( 9, PinMode.PWM );
+		arduino.pinMode( 10, PinMode.PWM );
+
+		//Now, I will set pin 13 to INPUT so that it enables reporting, and we'll be informed whenever the pin value changes.
+		arduino.pinMode( 13, PinMode.INPUT );
+
+		//next, I'm going to spin up two threads. One will write values to these pins, and the other will periodically read them
+		writerThreadShouldExit = false;
+		readerThreadShouldExit = false;
+		Task.Run( () => Read() );
+		Task.Run( () => Write() );
+	}
+
+	/*
+	 * This function will loop until writerThreadShouldExit is set to true. It will constantly write
+	 * a value in the range of [0, 127] to pin 9 and twice that value to pin 10.
+	 * 
+	 * Remember: there exists special logic in our version of StandardFirmata that will modify the value of pin 10 (HIGH or LOW) based on the value written to these pins.
+	 */
+	private void Write()
+	{
+		int pinValue = 0;
+		while( !writerThreadShouldExit )
+		{
+			if( pinValue > 127 ) pinValue = 0;
+
+			//this is our "critical section". Since pin 10 must always be twice the value of pin 9, we're going to lock
+			//before we send these messages. This guarantees that no other thread can use this UwpFirmata object while we 'own' the lock.
+			//it would be very bad if someone tried to read pin 13 in between our two messages!
+			firmata.@lock();
```

Comment:
The UwpFirmata object uses a mutex internally, rather than locking on itself. Best practice is to continue using the same lock